source/include/behaviour/cs_BehaviourHandler.h

```diff
@@ -62,6 +63,12 @@ class BehaviourHandler : public EventListener, public Component {
 	 */
 	bool requiresAbsence(Time t);
 
+	/**
+	 * Checks if the given behaviour is valid. Queries the presence handler and SystemTime for
+	 * the current state, and resolves the behaviour type to call isValid.
+	 */
+	bool validateBehaviour(Behaviour* behaviour) const;
```

Comment:
This confuses me: valid seems to mean 2 different things here: valid data, but also whether the behaviour is active under current conditions.